cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/metadata/EntityMetadataEmitterContext.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.api.metadata;
+
+import co.cask.cdap.api.metadata.MetadataEntity;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Interfaces for methods exposed in pipeline to write metadata. This writer is tied to an
+ * entity and can only write metadata for the specific entity or its derivatives
+ */
+public interface EntityMetadataEmitterContext {
```

Comment:
none of the methods here throws checked exceptions. How does the user know whether it was successful or not?